monitoring/backups_collector.go

```diff
@@ -0,0 +1,94 @@
+package monitoring
+
+import (
+	"strconv"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/longhorn/longhorn-manager/datastore"
+	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
+)
+
+type BackupsCollector struct {
+	*baseCollector
+
+	sizeMetric  metricInfo
+	stateMetric metricInfo
+}
+
+func NewBackupsCollector(
+	logger logrus.FieldLogger,
+	nodeID string,
+	ds *datastore.DataStore) *BackupsCollector {
+
+	vc := &BackupsCollector{
+		baseCollector: newBaseCollector(subsystemBackups, logger, nodeID, ds),
+	}
+
+	vc.sizeMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemBackups, "actual_size_bytes"),
+			"Actual space used by each backup of the snapshot on the corresponding node",
+			[]string{nodeLabel, backupsLabel},
+			nil,
+		),
+		Type: prometheus.GaugeValue,
+	}
+
+	vc.stateMetric = metricInfo{
+		Desc: prometheus.NewDesc(
+			prometheus.BuildFQName(longhornName, subsystemBackups, "stats_backup_status"),
+			"State of this backup",
+			[]string{nodeLabel, backupsLabel},
+			nil,
+		),
+		Type: prometheus.GaugeValue,
+	}
+
+	return vc
+}
+
+func (vc *BackupsCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- vc.sizeMetric.Desc
+	ch <- vc.stateMetric.Desc
+}
+
+func (vc *BackupsCollector) Collect(ch chan<- prometheus.Metric) {
+	defer func() {
+		if err := recover(); err != nil {
+			vc.logger.WithField("error", err).Warn("panic during collecting metrics")
+		}
+	}()
+
+	backupsLists, err := vc.ds.ListBackups()
+	if err != nil {
+		vc.logger.WithError(err).Warn("error during scrape ")
+		return
+	}
+
+	for _, v := range backupsLists {
+		if v.Status.OwnerID == vc.currentNodeID {
```

Comment:
I'm thinking that should we add this check because for the new Backup CR being created, the `status.OwnerID` is empty at that moment until the backup_controller handles the CR and updates the `status.OwnerID` field.
Therefore, within the short period, the Prometheus collector would not collect this Backup CR information.

Reply:
Because backups are cluster-wide concept, however our metric server is actually running in each manager. So, it seems we still need this condition to avoid polling duplicated info from different nodes.